### PR TITLE
feat: Use the actions to register and delete the push token

### PR DESCRIPTION
**Pull request by KaterinaWire**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

We have a `PushTokenStrategy` which is responsible for getting, sending and deleting a push token in BE. A case has been found where we try to delete a token that is not on the BE, we get an error and do not send a new token. As a result, we do not have a token on BE and the user doesn't receive notifications.

### Solutions

Remove `PushTokenStrategy` and use the actions to register and delete the push token introduced in the RS.

Needs releases with:

- [ wireapp/wire-ios-request-strategy/pull/517 ] 

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Source/SessionManager/SessionManager+Push.swift [hunk 1]
```diff
@@ -75,7 +75,7 @@ extension SessionManager: PKPushRegistryDelegate {
 
         // Delete push token from all running sessions.
         backgroundUserSessions.values.forEach { userSession in
-            userSession.deletePushKitToken()
+            userSession.deletePushKitToken(nil)
```
Comment (johnxnguyen) [inline Source/SessionManager/SessionManager+Push.swift:78]: I don't think the `nil` is needed.

File: Source/UserSession/ZMUserSession+Push.swift [hunk 2]
```diff
@@ -101,28 +102,58 @@ extension ZMUserSession {
...
+                    Logging.push.safePublic("Failed to register push token with backend: \(error)")
+                }
             }
+
+            action.send(in: syncMOC.notificationContext)
         }
     }
 
-    func deletePushKitToken(isLegacy: Bool = false) {
+    func deletePushKitToken(_ completion: (() -> Void)? = nil) {
```
Comment (johnxnguyen) [inline Source/UserSession/ZMUserSession+Push.swift:128]: Usually completion handlers have an argument label:

```suggestion
    func deletePushKitToken(completion: (() -> Void)? = nil) {
```

**Review by johnxnguyen — APPROVED** [review]

**Comment by KaterinaWire** [comment]
release major

**Comment by zenkins** [comment]
Hello, @KaterinaWire! Merging PR for command **release major** for PR #**1730**. Will create a **major** release after merged.
Next version is 424.0.0.

**Comment by KaterinaWire** [comment]
release major

**Comment by zenkins** [comment]
Hello, @KaterinaWire! Merging PR for command **release major** for PR #**1730**. Will create a **major** release after merged.
Next version is 424.0.0.

**Pull request merged**